### ISSUE TITLE
TimeSeriesKMeans with different length time series

### ISSUE THREAD

**Issue by MAGomes95** [rtavenar/tslearn#215]
Hello,

Without any delays, I will begin explaining the bug on TimeSeriesKMeans function.

At the moment, I am handling a dataset with several time series having different lengths.
When I try to cluster this time series by means of the dtw or soft-dtw metrics, and despite n_cluster parameter, the model predicts at max two clusters, in other situations, assigns all the time series to the same cluster.

Following I give an example of both situations

![bug_tslearn](https://user-images.githubusercontent.com/30867275/78295662-657a2e80-7524-11ea-826e-82616e8a6f6e.jpeg)

Thank you,

M.

**Comment by LiJinyang12** [comment]
Hi all, I ran into the same bug these days. When I tried to use TimeSeriesKMeans to cluster time series of variable length, besides the inproper clustering result, I also noticed that, if I applay fit() then predict(), a ValueError was raised: **ValueError: Dimensions (except first) must match!** . But if I applied fit_predict(), no error or warning happened. However, the clustering result did not match with n_cluster parameter. In addition, the cluster_centers_ are all of the length of the shorted serie in my dataset.

**Comment by rtavenar** [comment]
@LiJinyang12 your bug about dimensions not matching is probably rather related to #220

**Comment by rtavenar** [comment]
@MAGomes95 

A fix has been implemented and merged to `dev` branch. It will be available in tslearn v0.4.
Your example is tested and works OK (number of predicted clusters is equal to `n_clusters`). If you want to use it before v0.4 is out, you should first install the `dev` tslearn:

```bash
python -m pip install https://github.com/rtavenar/tslearn/archive/dev.zip
```

**Comment by ParthibanGowthamanON** [comment]
When we change the metric to euclidean in above example its throwing error, Is there any fix to use euclidean for different length series?

```
import os
os.chdir('C:/Users/Prathiban.gowthaman/Documents/GitRepos/CI_Multiple/disaggregation/data/processed')
from tslearn.clustering import TimeSeriesKMeans
from tslearn.utils import to_time_series_dataset
data=[[1,2,3],[7,8,9,11],[0.1,0.2,2],[1,1,1,9],[10,20,30,1000]]
data = to_time_series_dataset(data)
TimeSeriesKMeans(n_clusters=2,metric='euclidean').fit_predict(data)
```

<img width="534" alt="image" src="https://user-images.githubusercontent.com/93656255/152731108-dbd1a7c6-2df8-4f3d-b0f5-7345552c4848.png">
